applications: nrf_desktop: Add HID providers (#23090)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/hids.rst

.. note::
   The nRF Desktop centrals perform the GATT service discovery and reenable the HID notifications on every reconnection.
   A HID report that is received before the subscription is reenabled will be dropped before it reaches the application.

>    The :ref:`CONFIG_DESKTOP_HIDS_FIRST_REPORT_DELAY <config_desktop_app_options>` option is set to 1000 ms for nRF Desktop keyboards (:ref:`CONFIG_DESKTOP_PERIPHERAL_TYPE_KEYBOARD <config_desktop_app_options>`) to make sure that the input is not lost on reconnection with the nRF Desktop dongle.

Reviewer: ```suggestion
   The :ref:`CONFIG_DESKTOP_HIDS_FIRST_REPORT_DELAY <config_desktop_app_options>` option is set to 1000 ms for nRF Desktop keyboards (:ref:`CONFIG_DESKTOP_PERIPHERAL_TYPE_KEYBOARD <config_desktop_app_options>`) to ensure that input is not lost when reconnecting to the nRF Desktop dongle.
```

Author: This was already reviewed and merged under separate PR. I will rebase and drop it from here.
